umongo/data_objects.py

```diff
@@ -165,19 +165,28 @@ def __init__(self, document_cls, pk):
         self.pk = pk
         self._document = None
 
-    def fetch(self, no_data=False, force_reload=False):
+    def fetch(self, no_data=False, force_reload=False, projection=None):
         """
         Retrieve from the database the referenced document
 
-        :param no_data: if True, the caller is only interested in whether or
-            not the document is present in database. This means the
+        :param no_data: if True, the caller is only interested in whether
+            the document is present in database. This means the
             implementation may not retrieve document's data to save bandwidth.
         :param force_reload: if True, ignore any cached data and reload referenced
             document from database.
+        :param projection: if supplied, this is a dictionary or list describing
+            a projection which limits the data returned from database.
         """
         raise NotImplementedError
     # TODO replace no_data by `exists` function
```

Comment:
I guess we could deprecate `no_data`, then? (No pressure to do this in this PR.)

Reply:
Sounds good, this can be deprecated in a future PR/release.